src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -207,6 +252,14 @@ default boolean isEmpty() {
      */
     byte[] getBases();
 
+    /**
+     * @return The read sequence as ASCII bytes ACGTN=, or an empty byte[] if no sequence is present.
+     *
+     * This method DOES NOT make a defensive copy of the bases array before returning it, so modifying the
+     * returned array WILL alter the bases in the read. CALLER BEWARE!
+     */
+    byte[] getBasesNoCopy();
```

Comment:
I would strongly consider renaming these to getBasesUnsafe() and changing the description to make it clear that editing the bases only MAY alter the bases in the read, not WILL alter the bases in the read, since half of the implementations do not.  Then I would provide a default implementation that's just getBases()

Reply:
I don't like the name `getBasesUnsafe()`, since it scare people without explaining why they should be scared. I think it's better to have the name explain how it differs from `getBases()`.

I will change the description, however, to indicate that implementations will avoid a copy only if it's possible to do so.

Reply:
Also added default implementations.